File: account.c
#include <stdio.h>
#include <stdlib.h>
#include "utils.h"

void createAccount() {
    char accountName[100];
    printf("Entrez le nom du compte: ");
    scanf("%s", accountName);

    AccountInfo accountInfo;
    strcpy(accountInfo.accountName, accountName);
    accountInfo.Argent = 0;
    accountInfo.CarteBancaire = 0;
    accountInfo.Permis = 0;
    accountInfo.PhotosIdentite = 0;
    accountInfo.CartesVisite = 0;

    // Demander le nombre d'objets de chaque catégorie
    const char* categories[] = {"Argent", "CarteBancaire", "Permis", "PhotosIdentite", "CartesVisite"};

    for (int i = 0; i < sizeof(categories) / sizeof(categories[0]); i++) {
        accountInfo.values[i] = readLimitedNumber(categories[i], getCategoryLimit(categories[i]));
    }

    // Enregistrer la transaction complète dans le fichier de transactions
    char transaction[200];
    snprintf(transaction, sizeof(transaction), "Date=%s;Type=Création;Compte=%s;Argent=%d;CarteBancaire=%d;Permis=%d;PhotosIdentite=%d;CartesVisite=%d\n",
             getCurrentDate(), accountInfo.accountName, accountInfo.Argent, accountInfo.CarteBancaire, accountInfo.Permis, accountInfo.PhotosIdentite, accountInfo.CartesVisite);
    appendToFile(getTransactionFilePath(), transaction);

    printf("Compte créé avec succès!\n");
}

int getCategoryLimit(const char* category) {
    if (strcmp(category, "Argent") == 0) {
        return 50000;
    } else if (strcmp(category, "CarteBancaire") == 0) {
        return 10;
    } else if (strcmp(category, "Permis") == 0) {
        return 5;
    } else if (strcmp(category, "PhotosIdentite") == 0) {
        return 3;
    } else if (strcmp(category, "CartesVisite") == 0) {
        return 10;
    } else {
        return 0;
    }
}

int readLimitedNumber(const char* category, int limit) {
    int value;

    do {
        printf("Entrez l'unité initiale de %s (limite: %d) : ", category, limit);
        scanf("%d", &value);

        if (value > limit) {
            printf("La valeur ne doit pas dépasser la limite de %d. Veuillez réessayer.\n", limit);
        }
    } while (value > limit);

    return value;
}

File: account.js
const readlineSync = require('readline-sync');
const fs = require('fs');
const utils = require('./utils');

function createAccount() {
    const accountName = readlineSync.question('Entrez le nom du compte: ');
    //const initialBalance = parseFloat(readlineSync.question('Entrez le solde initial: '));

    const accountInfo = {
        accountName,
        Argent: 0,
        CarteBancaire: 0,
        Permis: 0,
        PhotosIdentite: 0,
        CartesVisite: 0,
    };

    // Demander le nombre d'objets de chaque catégorie
    const categories = ['Argent','CarteBancaire', 'Permis', 'PhotosIdentite', 'CartesVisite'];

    for (const category of categories) {
        accountInfo[category] = readLimitedNumber(category, getCategoryLimit(category));
    }

    // Enregistrer la transaction complète dans le fichier de transactions
    const transaction = `Date=${utils.getCurrentDate()};Type=Création;Compte=${accountName};Argent=${accountInfo['Argent']};CarteBancaire=${accountInfo['CarteBancaire']};Permis=${accountInfo['Permis']};PhotosIdentite=${accountInfo['PhotosIdentite']};CartesVisite=${accountInfo['CartesVisite']}\n`;
    fs.appendFileSync(utils.getTransactionFilePath(), transaction);

    console.log('Compte créé avec succès!');
}

// Fonction utilitaire pour obtenir la limite pour chaque catégorie
function getCategoryLimit(category) {
    switch (category) {
        case 'Argent':
            return 50000;
        case 'CarteBancaire':
            return 10;
        case 'Permis':
            return 5;
        case 'PhotosIdentite':
            return 3;
        case 'CartesVisite':
            return 10;
        default:
            return 0;
    }
}

function readLimitedNumber(category) {
    const limit = getCategoryLimit(category);
    let value;

    do {
        value = parseInt(readlineSync.question(`Entrez l'unité initiale de ${category} (limite: ${limit}) : `), 10) || 0;

        if (value > limit) {
            console.log(`La valeur ne doit pas dépasser la limite de ${limit}. Veuillez réessayer.`);
        }
    } while (value > limit);

    return value;
}


module.exports = {
    getCategoryLimit,
    readLimitedNumber,
    createAccount
};
